src/submission/exceptions/exceptions.ts

```diff
@@ -57,15 +58,21 @@ const validateFieldValueWithExceptions = ({
   schemaName,
   fieldValue,
   validationErrorFieldName,
+  valueType,
 }: {
   record: DeepReadonly<TypedDataRecord>;
   programException: ProgramException | null;
   entityException: EntityException | null;
   schemaName: ClinicalEntitySchemaNames;
-  fieldValue: string;
+  fieldValue: string | undefined;
   validationErrorFieldName: string;
+  valueType?: dictionaryEntities.ValueType;
 }): boolean => {
-  const allowedValues: Set<string> = new Set();
+  const allowedValues: Set<string | undefined> = new Set();
+
+  const isNumericField =
+    valueType === dictionaryEntities.ValueType.INTEGER ||
+    valueType === dictionaryEntities.ValueType.NUMBER;
```

Comment:
this would be a nice reusable function:

```ts
const isNumericField = (valueType: string): boolean => 
  valueType === dictionaryEntities.ValueType.INTEGER ||
  valueType === dictionaryEntities.ValueType.NUMBER;
```